File: CodeForces/80a/main.cpp
#include <cmath>
#include <iostream>

bool isPrime(int n) {
  if (n <= 1) {
    return false;
  }

  for (int i = 2; i < int(pow(n, 0.5)) + 1; i++) {
    if (n % i == 0) {
      return false;
    }
  }
  return true;
}

int main() {

  int a, b;
  std::cin >> a >> b;

  if (isPrime(a) && isPrime(b)) {
    int c = 0;
    for (int x = a + 1; x <= b; x++) {
      if (isPrime(x))
        c++;
    }
    if (c == 1) {
      std::cout << "YES" << std::endl;
    } else {
      std::cout << "NO" << std::endl;
    }
  } else {
    std::cout << "NO" << std::endl;
  }

  return 0;
}

File: CodeForces/80a/main.py
def inp():
    return int(input())


def inlt():
    return list(map(int, input().split()))


def isPrime(x):
    if x <= 1:
        return False

    for i in range(2, int(x**0.5) + 1):
        if x % i == 0:
            return False

    return True


a, b = inlt()
c = 0

if isPrime(a) and isPrime(b):
    for i in range(a + 1, b + 1):
        if isPrime(i):
            c += 1

    if c == 1:
        print("YES")
    else:
        print("NO")

else:
    print("NO")
